packages/ui/src/components/DataStory/clients/WorkspacesApi.ts

```diff
@@ -0,0 +1,14 @@
+export interface Tree {
+  path: string;
+  type: 'file' | 'folder';
+  content?: string;
+  children?: Tree[];
+}
+
+export interface WorkspacesApi {
+  get: ({ path }) => Promise<Tree>
+  create: ({ path, tree }: { path: string, tree: Tree }) => Promise<Tree>;
+  update: ({ path, tree }: { path: string, tree: Tree }) => Promise<Tree>
```

Comment:
Does the `update` action merge or overwrite existing content?

Reply:
Good question! Lets get back to this.